File: 23-05-2026/string_zipper.py
def zip_strings(a, b):
    aSplit, bSplit, finalStr = list(a), list(b), "";
    remainingB = "".join(bSplit[len(aSplit):]);

    for i in range(len(aSplit)):
        finalStr += aSplit[i] + bSplit[i];

    return finalStr + remainingB

print(zip_strings("abc", "123")); # "a1b2c3".
print(zip_strings("acegikmoqsuwy", "bdfhjlnprtvxz")); # "abcdefghijklmnopqrstuvwxyz".
print(zip_strings("day", "night")); # "dnaiyght".
print(zip_strings("python", "javascript")); # "pjyatvhaosncript".
print(zip_strings("feCdCm", "reoeap")); # "freeCodeCamp".

File: 23-05-2026/string_zipper.js
function zipStrings(a, b) {
  let [aSplit, bSplit, finalStr] = [a.split(""), b.split(""), ""];
  const remainingB = bSplit?.slice(aSplit.length).join("") || "";

  aSplit.forEach((elem, i) => (finalStr += elem + bSplit[i]));

  return finalStr.concat(remainingB);
}

console.log(zipStrings("abc", "123")); // "a1b2c3".
console.log(zipStrings("acegikmoqsuwy", "bdfhjlnprtvxz")); // "abcdefghijklmnopqrstuvwxyz".
console.log(zipStrings("day", "night")); // "dnaiyght".
console.log(zipStrings("python", "javascript")); // "pjyatvhaosncript".
console.log(zipStrings("feCdCm", "reoeap")); // "freeCodeCamp".
